gplugins/lumerical/write_sparameters_lumerical.py

```diff
@@ -550,12 +518,120 @@ def _sample_convergence_wavelength():
     # component = gf.components.straight(length=2.5)
     component = gf.components.mmi1x2()
 
-    material_name_to_lumerical = dict(si=(3.45, 2))  # or dict(si=3.45+2j)
+    material_name_to_lumerical = dict(
+        si="Si (Silicon) - Palik",
+        substrate="Si (Silicon) - Palik",
+        box="SiO2 (Glass) - Palik",
+        clad="SiO2 (Glass) - Palik",
+    )  # or dict(si=3.45+2j)
+
+    # Create layerstack
```

Comment:
is this a dummy layersatck? regardless, it should be in its own file- maybe this can be combined with the one used in the test? it's a lot of lines...

Reply:
Yes, it's a dummy layerstack with no doping and fixes the original layerstack in the generic PDK by specifying the clad and box layers as sio2 material with layer_type = "background". It was originally set to "grow", which prevents those layers from being generated properly in Lumerical.

This is all test code anyways so shall we remove it completely?

Reply:
is this the same as `layerstack_lumerical2023` ? I'm not quite sure what this is achieving- to confirm, this whole fie will be removed when this functionality is migrated to the `fdtdDesignRecipe`, right?

Reply:
Agreed, it's test code and can be removed.

> this whole fie will be removed when this functionality is migrated to the fdtdDesignRecipe, right?


Partially yes, the write_sparameters_lumerical is the combination of simulation features (setup, running, extraction) and we will separate this into:
1. fdtd.py
LumericalFdtdSimulation class (simulation setup and data extraction/visualization features)
2. Under lumerical/design_recipe directory
FdtdDesignRecipe (specifies design intent) that invokes functions from LumericalFdtdSimulation and performs additional data processing to form compact models or contains optimizations
3. simulation_settings.py
FdtdSimulationSetup (Pydantic Base Model)
4. convergence_settings.py
FdtdConvergenceSetup (Pydantic Base Model)